lib/PhasarLLVM/DataFlowSolver/IfdsIde/Problems/IDESecureHeapPropagation.cpp

```diff
@@ -305,9 +314,9 @@ IDESecureHeapPropagation::SHPGenEdgeFn::getInstance(l_t Val) {
   static std::array<std::shared_ptr<SHPGenEdgeFn>, 3> Cache = {nullptr, nullptr,
                                                                nullptr};
   auto Ind = static_cast<std::underlying_type<l_t>::type>(Val);
-  if (!Cache[Ind]) {
-    Cache[Ind] = std::make_shared<SHPGenEdgeFn>(Val);
+  if (!Cache.at(Ind)) {
+    Cache.at(Ind) = std::make_shared<SHPGenEdgeFn>(Val);
```

Comment:
```suggestion
    Cache[Ind] = std::make_shared<SHPGenEdgeFn>(Val);
```
If we already did a bounce check in the previous line, this one should not be necessary.